### ISSUE TITLE
Can't handle requests with array body

### ISSUE THREAD

**Issue by bradennapier** [elastic/apm-agent-nodejs#1360]
**Describe the bug**

When an error occurs and the request body is an array, it provides error:

```
I[#/context/request/body] S[#/allOf/1/properties/context/properties/request/properties/body/type] expected object or string or null, but got array
```

This also seems to produce these errors as well:

```
15|api      | APM Server transport error (400): Unexpected APM Server response
15|api      | APM Server accepted 25 events in the last request
```

**To Reproduce**

Generate an error when the request body is an array.

**Expected behavior**

Should support the valid situation when a request is an array

**Environment (please complete the following information)**

- OS: [e.g. Linux] OSX
- Node.js version: 10.15.1
- APM Server version: 7.3.1
- Agent version: 2.16.2

**How are you starting the agent? (please tick one of the boxes)**

- [ ] Calling `agent.start()` directly (e.g. `require('elastic-apm-node').start(...)`)
- [x] Requiring `elastic-apm-node/start` from within the source code
- [ ] Starting node with `-r elastic-apm-node/start`

**Comment by Qard** [comment]
Appears to be due to this logic: https://github.com/elastic/apm-agent-nodejs/blob/master/lib/parsers.js#L110-L128

It seems to use a stringified form _only_ when the length exceeds the limit and thus gets truncated.

@watson Shouldn't that _always_ get converted to a string? Or is it _intentional_ that it keeps object form when it's small enough?

**Comment by watson** [comment]
The idea was to be able to display the body in a nicer way in the APM UI. And since the `request.body` field isn't indexed in Elasticsearch there shouldn't any overhead in storing it in a more usable format.

I just checked [the allowed types](https://github.com/elastic/apm-server/blob/88f3ec674aa1bd0698ccf5950cc5dfe765f01fd7/docs/spec/request.json#L7-L10), and the APM Server only allows `string`, `object`, or `null`. To be complete it would need to support `number`, `boolean`, and `array` as well. But if at all, this would not make it in until v7.5 of the APM Server, so we'd need to deal with these cases in the agent for now.

So the solution currently is to either treat `null` and `object` in a special way and stringify everything else, or we should just stringify everything - including `null` and `object`.

@Qard What do you think?

**Comment by Qard** [comment]
Maybe just _always_ stringify for now. Easier and less error-prone that way. We can re-evaluate later when APM Server supports the other forms better.

**Comment by watson** [comment]
Sound good to me.

**Comment by watson** [comment]
A fix for this has now been released in v2.17.1. Thanks a lot again for reporting this issue 💯

**Comment by Sonosar** [comment]
Is there a way to parse this value back to json in APM server?
